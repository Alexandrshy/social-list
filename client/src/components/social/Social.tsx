import React from "react";

import { Icon } from "../icon/Icon";

import { TYPE_USE, SIZE_MEDIUM } from "../icon/constants";

import s from "./Social.module.css";

export const SOCIAL_LIST = [
  {
    name: "twitter",
    link: "/"
  },
  {
    name: "facebook",
    link: "/"
  },
  {
    name: "telegram",
    link: "/"
  }
];

export const Social: React.FC = () => {
  return (
    <div className={s.social}>
      <ul className={s.list}>
        {SOCIAL_LIST.map(({ name, link }, index) => {
          return (
            <li className={s.item} key={index}>
              <a href={link} className={s.link}>
                <Icon name={name} size={SIZE_MEDIUM} type={TYPE_USE} />
              </a>
            </li>
          );
        })}
      </ul>
    </div>
  );
};
