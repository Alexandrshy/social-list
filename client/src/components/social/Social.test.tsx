import React from "react";
import renderer from "react-test-renderer";

import { Social } from "./Social";

jest.mock("../icon/Icon", () => ({
  Icon: "Icon"
}));

describe("Social component", () => {
  it("should render", () => {
    const tree = renderer.create(<Social />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
