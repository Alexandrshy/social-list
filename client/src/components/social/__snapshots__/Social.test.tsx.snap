// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social component should render 1`] = `
<div
  className="social"
>
  <ul
    className="list"
  >
    <li
      className="item"
    >
      <a
        className="link"
        href="/"
      >
        <Icon
          name="twitter"
          size="medium"
          type="use"
        />
      </a>
    </li>
    <li
      className="item"
    >
      <a
        className="link"
        href="/"
      >
        <Icon
          name="facebook"
          size="medium"
          type="use"
        />
      </a>
    </li>
    <li
      className="item"
    >
      <a
        className="link"
        href="/"
      >
        <Icon
          name="telegram"
          size="medium"
          type="use"
        />
      </a>
    </li>
  </ul>
</div>
`;
